src/app/parts-editor/components/dialog-database/dialog-database.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { FormBuilder, FormGroup, FormArray } from '@angular/forms';
+import { PartWorkbook } from '../../models/editor';
+import { Subject } from 'rxjs';
+import { Cell } from '../../models/cell';
+
+@Component({
+  selector: 'app-dialog-database',
+  templateUrl: './dialog-database.component.html',
+  styleUrls: ['./dialog-database.component.scss'],
+})
+export class DialogDatabaseComponent implements OnInit, OnDestroy {
+  destroy$ = new Subject<void>();
+  missingPartsForm: FormGroup;
+
+  constructor(
+    public dialogRef: MatDialogRef<DialogDatabaseComponent>,
+    private formBuilder: FormBuilder,
+    @Inject(MAT_DIALOG_DATA) public missingParts: PartWorkbook
+  ) {}
+
+  ngOnInit() {
+    this.missingPartsForm = this.formBuilder.group({
```

Comment:
The top group is pointless. There is only one property there, just use array as a top level